Source/Forms/Models/Gen3/EggModel3.cpp

```diff
@@ -62,7 +62,14 @@ QVariant EggModel3::data(const QModelIndex &index, int role) const
         {
         case 0:
         case 1:
-            return state.getGenerateAdvance();
+            switch (method)
+            {
+            case Method::RSBred:
+            case Method::FRLGBred:
+                return state.getGenerateAdvance();
+            default:
+                return state.getAdvances();
+            }
```

Comment:
I am not a big fan of this approach. It would be best to modify getColumn and the QStringList holding all the strings if necessary. This way we only do the switch statement once.

Reply:
> I am not a big fan of this approach. It would be best to modify getColumn and the QStringList holding all the strings if necessary. This way we only do the switch statement once.

Changed